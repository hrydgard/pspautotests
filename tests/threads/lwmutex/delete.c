#include "shared.h"

#define DELETE_TEST(title, workarea) { \
	int result = sceKernelDeleteLwMutex(workarea); \
	if (result == 0) { \
		printf("%s: OK\n", title); \
	} else { \
		printf("%s: Failed (%X)\n", title, result); \
	} \
}

int main(int argc, char **argv) {
	SceLwMutexWorkarea workarea, workarea1, workarea2;

	sceKernelCreateLwMutex(&workarea, "delete", 0, 0, NULL);

	DELETE_TEST("Normal", &workarea);
	// Crash.
	//DELETE_TEST("NULL", 0);
	DELETE_TEST("Invalid", (void*) 0xDEADBEEF);
	DELETE_TEST("Deleted", &workarea);

	sceKernelCreateLwMutex(&workarea1, "delete", 0, 0, NULL);
	memcpy(&workarea2, &workarea1, sizeof(SceLwMutexWorkarea));
	sceKernelDeleteLwMutex(&workarea1);
	DELETE_TEST("Copy", &workarea2);

	FAKE_LWMUTEX(workarea, 0, 0);
	DELETE_TEST("Fake", &workarea2);

	BASIC_SCHED_TEST("Delete other",
		result = sceKernelDeleteLwMutex(&workarea2);
	);
	BASIC_SCHED_TEST("Delete same",
		result = sceKernelDeleteLwMutex(&workarea1);
	);
	BASIC_SCHED_TEST("Invalid",
		result = sceKernelDeleteLwMutex((void*) 0xDEADBEEF);
	);

	return 0;
}