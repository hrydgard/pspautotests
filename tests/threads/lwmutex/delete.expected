Normal: OK
Invalid: Failed (800200D3)
Deleted: Failed (800201CA)
Copy: Failed (800201CA)
Fake: Failed (800201CA)
Delete other: A1B1C2E1D1F2 (thread=800201B5, main=00000000)
Delete same: A1B1C2D1E2F1 (thread=800201B5, main=00000000)
Invalid: A1B1C2E1D1F2 (thread=800201B5, main=800200D3)
